### PR TITLE
Remove deprecated and orphaned generated code

### PR DESCRIPTION
**Pull request by thockin**
All of this code is marked as being generated but is not regenerated by any tool I can find.

/kind cleanup
/sig architecture

```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/client-go/applyconfigurations/meta/v1/groupversionkind.go [hunk 1]
```diff
@@ -1,57 +0,0 @@
-/*
```
Comment (liggitt) [inline staging/src/k8s.io/client-go/applyconfigurations/meta/v1/groupversionkind.go:1]: fyi @jpbetz ... if everything builds without these applyconfigurations for non-top-level meta/v1 types, then they were unused, right?

Comment (jpbetz) [reply staging/src/k8s.io/client-go/applyconfigurations/meta/v1/groupversionkind.go:1]: That is correct. I've reviewed all the unused applyconfigurations.

A bunch came from names of validating policy types that we didn't end up using in our final design. This demonstrates that during initial API development, when names of things change, it is easy to end up with orphaned generated fields that get merged. This is hard to catch during review due to the sheer volume of generated code.

Even most of the meta/v1 generate files appeared as part of the validating policy changes ([relevant commit](https://github.com/kubernetes/kubernetes/commit/19b2df871560816c3eb29fbb271953afab1e39c8#diff-43f3798c7c7c5c006f80cf5236a5a05e21c2f21e956185d8c16ea907e0d750f3)).

Deleting these files is safe.

**Comment by jpbetz** [comment]
/hold cancel

Removal of applyconfiguration is safe. See https://github.com/kubernetes/kubernetes/pull/115186#discussion_r1081544865

/lgtm

**Pull request merged**